internal/core/src/storage/ChunkCache.cpp

```diff
@@ -0,0 +1,106 @@
+// Licensed to the LF AI & Data foundation under one
+// or more contributor license agreements. See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership. The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "ChunkCache.h"
+#include "Exception.h"
+
+namespace milvus::storage {
+
+std::shared_ptr<ColumnBase>
+ChunkCache::Read(const std::string& filepath) {
+    auto path = std::filesystem::path(path_prefix_) / filepath;
+
+    ColumnTable::const_accessor ca;
+    if (columns_.find(ca, path)) {
+        return ca->second;
+    }
+    ca.release();
+
+    auto object_data =
+        GetObjectData(cm_.get(), std::vector<std::string>{filepath});
+    AssertInfo(object_data.size() == 1, "GetObjectData failed");
+    auto field_data = object_data[0];
+
+    auto column = Mmap(path, field_data);
+    columns_.emplace(path, column);
+    return column;
+}
+
+void
+ChunkCache::Remove(const std::string& filepath) {
+    auto path = std::filesystem::path(path_prefix_) / filepath;
+    columns_.erase(path);
+}
+
+void
+ChunkCache::Prefetch(const std::string& filepath) {
+    auto path = std::filesystem::path(path_prefix_) / filepath;
+    ColumnTable::const_accessor ca;
+    if (!columns_.find(ca, path)) {
+        return;
+    }
+    auto column = ca->second;
+    auto ok =
+        madvise(reinterpret_cast<void*>(const_cast<char*>(column->Data())),
+                column->ByteSize(),
+                MADV_SEQUENTIAL);
```

Comment:
we'd better to test the performance with this, as `MADV_SEQUENTIAL` aggressively reads ahead and may free the mapping soon. The access pattern of columns is sequential right, but it's random in reduce stage. `MADV_WILLNEED` and `MADV_RANDOM` should be also options.